source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,317 @@
+type for each option.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Node.js Option
+     - Driver Option
+     - Description
+
+   * - ``ca``, type: ``Buffer``
+     - | ``tlsCAFile``, type: ``string``
+       | ``ca``, type: ``Buffer``
+     - Specifies either a single or bundle of certificate authorities to
+       trust when making a TLS connection.
+
+   * - ``cert``, type: ``Buffer``
+     - | ``tlsCertificateFile``, type: ``string``
+       | ``tlsCertificateKeyFile``, type: ``string``
+       | ``cert``, type: ``Buffer``
+     - Specifies the client certificate file or the client
+       private key file.
+
+   * - ``key``, type: ``Buffer``
+     - | ``tlsCertificateKeyFile``, type: ``string``
+       | ``key``, type: ``Buffer``
+     - Specifies the client private key file.
+
+   * - ``passphrase``, type: ``string``
+     - ``tlsCertificateKeyFilePassword``, type: ``string``
+     - Specifies the password to decrypt the client private key.
+
+.. note:: Additional TLS Client Options
+   
+   For **testing purposes**, you can set the ``tlsAllowInvalidHostnames`` and
+   ``tlsAllowInvalidCertificates`` client options.
+   
+   Setting the ``tlsAllowInvalidHostnames`` option to ``true`` disables
+   hostname verification.
+   Setting the ``tlsAllowInvalidCertificates`` to ``true`` disables
+   certificate validation and allows you to test a connection with an
+   invalid certificate, such as a self-signed certificate.
+   
+   Specifying either of these options in a production environment puts
+   your application at risk of becoming
+   `insecure <https://tlseminar.github.io/docs/mostdangerous.pdf>`__ or
+   vulnerable to expired certificates and to foreign processes posing
+   as valid client instances.
+
+For a full list of client options, see :ref:`node-connection-options`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions`` so
+that the Server can validate them before the client connects. The
+driver provides multiple TLS-related client options to reference your
+certificates. You can connect with the following option configurations:
+
+- Passing a ``SecureContext`` object to the ``secureContext`` option *(Recommended)*
+- Passing filepath strings to the ``tlsCAFile`` and
+  ``tlsCertificateKeyFile`` options
+- Passing ``Buffer`` objects that contain the contents of your
+  certificates to the ``ca``, ``cert``, and ``key`` options
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Connect
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to connect with
+the TLS protocol. By using a ``SecureContext`` object to store your
+certificates, you use the Node.js framework for TLS that is independent of the
+driver. When you use this option, you keep all TLS options in a
```

Comment:
Q: is the independence an advantage or just a side effect?

Reply:
I think it is an advantage, since it protects you from any changes that may occur to the driver functionality.